### ISSUE TITLE
Better folder structure

### ISSUE THREAD

**Issue by blue-atom** [aimacode/aima-python#858]
In this repo aima-python, every code is just residing at root folder. I think it will be good if there is some directory structure, for example, all the .ipynb files should be under the folder "/root/notebooks/"  and all the data structure classes should be under "/datastructures/" folder. I think it will be much more convenient to find files that way.

**Comment by ad71** [comment]
It would be convenient to find files, but import statements have to be changed for all files which is something @norvig doesn't want to do as implied by #803 . I don't think this will be a good idea. It's up to @norvig though.

**Comment by blue-atom** [comment]
@ad71 thanks for pointing out the previous issue.

**Comment by lalitmee** [comment]
@ad71 I know that changing the `import` path is the main problem if we want to have a nice project structure. But I think we can do it. It's not impossible. :innocent:

**Comment by ad71** [comment]
It can be done easily. Dr. Norvig apparently doesn't _want_ to.

**Comment by reachtarunhere** [comment]
Further another reason we kept the notebook in the root has to do with Binder. At that time it wasn't possible to get the imports right and make it work due to lack of config options. I am not sure if it has changed with the new version though.

**Comment by ad71** [comment]
In the current state, imports from other folders are handled by appending `__dirname__` to the imports using the `os.path` module. This works fine and is already done for the files in the GUI folder.

**Comment by lalitmee** [comment]
ohh.. if this is the scenario, then it should be your call to implement or not. I was just giving a suggestion. I think if you don't want to do, we can close the issue otherwise more people will come and say the same thing again. :innocent:

**Comment by rshrc** [comment]
Yeah, I saw the same scattered out project structure, this issue is still open though? @ad71 is work going on the restructuring?

**Comment by ad71** [comment]
Sure. Thanks!

**Comment by rshrc** [comment]
Since we are not restructuring the project structure of aima-python, there are many issues that need addressing. aimacode/aima-python#772 aimacode/aima-python#858 @MrDupin

**Comment by MrDupin** [comment]
I would say this is up to @norvig. I would like to see a structure as suggested [here](https://github.com/aimacode/aima-python/issues/772#issuecomment-373430188) by @ad71, but it is not up to me.

**Comment by rajatjain1997** [comment]
While we are not restructuring the repo, I think we should at least have a one-to-one correspondence on the "List of Notebooks" section with the chapters in the book. We could do this by mentioning the chapters that each notebook covers against their names. It'll make it easier for students to find the notebooks they need.

Something like this is already being attempted at aimacode-js: http://aimacode.github.io/aima-javascript/

**Comment by dmeoli** [comment]
Folder-structure discussion (2018); stale with no converged plan. Closing as archival.
